.github/workflows/update-release-version-template.yaml

```
@@ -0,0 +1,38 @@
+---
+name: Reusable Template to Update Release
+
+on:
+  workflow_call:
+    inputs:
+      release:
+        type: string
+        required: true
+
+jobs:
+  update-release:
+    name: Update Release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      - id: latest-release
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          echo version="$(gh release list -R etcd-io/etcd --json 'tagName' -q 'map(select(.tagName | test("^'"$(echo ${{ inputs.release }}. | sed s/\\./\\\./g)"'"))) | .[0].tagName')" >> "$GITHUB_OUTPUT"
+      - env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          minor=${{ inputs.release }}
+          index_file="content/en/docs/$minor/_index.md"
+          latest_version="${{ steps.latest-release.outputs.version }}"
+          branch="release-$minor-update-latest-release-to-$latest_version"
+          if ! grep git_version_tag "$index_file" | grep -v -e "$latest_version"'$'; then exit 0; fi
+          git checkout -b "$branch"
+          sed -i 's/git_version_tag:\sv\([0-9]\+\.\)\{2\}[0-9]\+/git_version_tag: '$latest_version'/' "$index_file"
+          git add "$index_file"
```

Nit: I think shellcheck suggests encapsulating variables as follows:

Perhaps just put the script standalone in a file and run shellcheck against it to double check.

```suggestion
          git add "${index_file}"
```

---

Hey @jmhbnz, thanks for the suggestion. I wasn't also sure if I wanted to place the script straight in this workflow file, or in a bash script in the repository. However, we don't have any other scripts like this in the repo. Would you recommend creating a new `scripts` directory (I don't know if this will break docsy and how we build the website)? I noticed that we have a `Makefile`, but I feel like this script is also particular to this action and may not even work outside of the workflow context. Let me know your thoughts so we can move forward with this :)

---

Initial reaction is I'm happy for this chunk of bash to stay in the workflow only for now. We can do it in iterations.

With that said - If you wanted to make a follow-up issue for moving it into a Makefile recipe perhaps called `update-etcd-version` or something similar then this makes sense to me. It would be nice to be able to run locally to test the script. It would just depend on `gh auth status` of local user which we may need to check for.

---

I initially moved the script to the `Makefile`. But, later, I moved it to its own script file. It was getting messy as an embedded script in the `Makefile`, and it wasn't possible to `shellcheck` it later. If we have more scripts later, we can add them as well.